File: sources/net.sf.j2s.core/test/src/com/falstad/DecentScrollbar.java
package com.falstad;

import a2s.Canvas;
import a2s.Scrollbar;
import java.awt.event.MouseEvent;
import java.awt.event.MouseListener;
import java.awt.event.MouseMotionListener;
import java.awt.Dimension;
import java.awt.Insets;
import java.awt.Color;
import java.awt.FontMetrics;
import java.awt.Event;
import java.awt.LayoutManager;
import java.awt.Image;
import java.awt.Container;
import java.awt.Component;
import java.awt.Graphics;
import java.awt.event.AdjustmentEvent;
import java.awt.event.AdjustmentListener;

// this is a scrollbar that notifies us when the user is _done_ fiddling
// with the value.
class DecentScrollbar extends Scrollbar implements AdjustmentListener {
    int value, lo, hi;
    DecentScrollbarListener listener;
    DecentScrollbar(DecentScrollbarListener parent, 
		    int start, int lo_, int hi_) {
	super(HORIZONTAL, start, 1, lo_, hi_);
	value = start;
	lo = lo_;
	hi = hi_;
	listener = parent;
	addAdjustmentListener(this);
    }
    @Override
    public void adjustmentValueChanged(AdjustmentEvent e) {
	if (getValueIsAdjusting())
	    listener.scrollbarValueChanged(this);
	else
	    listener.scrollbarFinished(this);
    }
    
};

File: sources/net.sf.j2s.core/test/src/com/falstad/DecentScrollbar.js
(function(){var P$=Clazz.newPackage("com.falstad"),I$=[];
var C$=Clazz.newClass(P$, "DecentScrollbar", null, 'a2s.Scrollbar', 'java.awt.event.AdjustmentListener');

C$.$clinit$ = function() {Clazz.load(C$, 1);
}

Clazz.newMeth(C$, '$init0$', function () {
var c;if((c = C$.superclazz) && (c = c.$init0$))c.apply(this);
this.value = 0;
this.lo = 0;
this.hi = 0;
this.listener = null;
}, 1);

Clazz.newMeth(C$, '$init$', function () {
}, 1);

Clazz.newMeth(C$, 'c$$com_falstad_DecentScrollbarListener$I$I$I', function (parent, start, lo_, hi_) {
C$.superclazz.c$$I$I$I$I$I.apply(this, [0, start, 1, lo_, hi_]);
C$.$init$.apply(this);
this.value = start;
this.lo = lo_;
this.hi = hi_;
this.listener = parent;
this.addAdjustmentListener$java_awt_event_AdjustmentListener(this);
}, 1);

Clazz.newMeth(C$, 'adjustmentValueChanged$java_awt_event_AdjustmentEvent', function (e) {
if (this.getValueIsAdjusting()) this.listener.scrollbarValueChanged$com_falstad_DecentScrollbar(this);
 else this.listener.scrollbarFinished$com_falstad_DecentScrollbar(this);
});

Clazz.newMeth(C$);
})();
//Created 2017-12-17 19:28:01
